.github/actions/generic/image-actions/common.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh -l
+
+#
+# this file looks at the first argument to determine whether to run the
+# node-os-tests.sh script or the publish-bindings.sh script.
+#
+script_name=$1
+script_to_run="/image-scripts/$1"
+
+if [ ! -f "$script_to_run" ] || [ ! -x "$script_to_run" ]; then
+    if [ "$script_to_run" != "echo" ]; then
+        echo "? error: not an executable file: ${script_to_run}"
+        exit 1
+    fi
+fi
+
+if [ "$script_name" = "node-os-tests.sh" ]; then
+    if [ $# -lt 5 ]; then
+        echo "? error: node-os-tests.sh requires 5 arguments, not $#"
+        echo "args: $*"
+        exit 1
+    fi
+elif [ "$script_name" = "publish.sh" ]; then
+    if [ $# -ne 8 ]; then
+        echo "? error: publish.sh requires 8 arguments, not $#"
+        echo "args: $*"
+        exit 1
+    fi
+else
```

Comment:
not sure if I'm missing something but I can only find 4 resp. 7 args in those scripts. I don't think $0 counts.

Reply:
got this during our conversation.